src/components/plots/piePlot.ts

```diff
@@ -37,6 +37,14 @@ export module Plot {
         _Util.Methods.warn("Only one dataset is supported in Pie plots");
         return this;
       }
+      var keyAndData = super._getKeyAndData(keyOrDataset, dataset);
+      // Check to see whether data is a Dataset or an array of values
+      var data = "_data" in keyAndData.data ? keyAndData.data._data : keyAndData.data;
+      data.forEach((datum) => {
+        if (datum.value < 0) {
+          _Util.Methods.warn("Negative values will not render correctly in a pie chart.");
```

Comment:
I think this is the wrong place to check this. There's no guarantee that the `value` property of each datum is actually driving the `value` projection. Additionally, even if it was, there's no guarantee that the `project()` call was passed an accessor that returns negative values.